### PR TITLE
pachctl: add a "misc grpc" command to make arbitrary RPCs

### PR DESCRIPTION
**Pull request by jrockway**
This allows you to call any RPC with any arguments you want:

Query a worker status:
```
$ pachctl misc grpc pachyderm.worker.Worker.Status --address="passthrough:///localhost:1080"
{"workerId":"default-edges-v1-rmwrv"}
```

Upload a file:
```
$ pachctl misc grpc pfs_v2.API.ModifyFile '{"set_commit":{"repo":{"name":"images","project":{"name":"default"},"type":"user"},"branch":{"repo":{"name":"images","type":"user","project":{"name":"default"}},"name":"master"}}}' '{"add_file":{"path":"/test.txt","raw":"aGVsbG8K"}}'
All messages sent
{}
```

List repos and format with jq:
```
$ pachctl misc grpc pfs_v2.API.ListRepo '{"projects":[{"name":"default"}]}' | jq '{name: .repo.name, description}' -c
{"name":"images","description":null}
{"name":"foo","description":null}
{"name":"bar","description":null}
{"name":"edges","description":"Output repo for pipeline default/edges."}
{"name":"edges","description":"Spec repo for pipeline default/edges."}
{"name":"edges","description":"Meta repo for pipeline edges"}
```

I have already found 2 bugs with this 😂 

Finally, I improved `log.Proto` to cleanly marshal things like `dynamicpb.Message`, so that debug logs `pachctl misc grpc ... -v` look normal, even though we aren't using the proto objects that have MarshalLogObject.

Example:

```
$ pachctl misc grpc pfs_v2.API.ModifyFile '{"set_commit":{"repo":{"name":"images","project":{"name":"default"},"type":"user"},"branch":{"repo":{"name":"images","type":"user","project":{"name":"default"}},"name":"master"}}}' '{"add_file":{"path":"/test.txt","raw":"aGVsbG8K"}}' -v
...
2023-08-31T14:01:51.715-0400    DEBUG   grpcClient.admin_v2.API/InspectCluster  admin_v2.API/InspectCluster: span start
2023-08-31T14:01:51.715-0400    DEBUG   grpcClient.admin_v2.API/InspectCluster  request {"request": {"client_version": {"major": 0, "minor": 0, "micro": 0, "additional": "", "git_commit": "", "git_tree_modified": "", "build_date": "", "go_version": "go1.21.0", "platform": "amd64"}, "current_project": {"name": "default"}}, "metadata": {"authn-token": "[MASKED]", "command": "..."}}
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.admin_v2.API/InspectCluster  admin_v2.API/InspectCluster: span finished ok   {"spanDuration": "8.477999ms", "headers": {"content-type": "application/grpc", "date": "Thu, 31 Aug 2023 18:01:51 GMT", "server": "envoy", "x-envoy-upstream-service-time": "7"}, "trailer": {}, "reply": {"id": "976ef26581fc45f8ad963f685cb34b3c", "deployment_id": "dev", "warnings_ok": true, "warnings": [], "proxy_host": "192.168.1.70", "proxy_tls": false, "paused": false, "web_resources": {"archive_download_base_url": "http://192.168.1.70/archive/", "create_pipeline_request_json_schema_url": "http://192.168.1.70/jsonschema/pps_v2/CreatePipelineRequest.schema.json"}}}
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile pfs_v2.API/ModifyFile: span start
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile stream started  {"metadata": {"authn-token": "[MASKED]", "command": "..."}}
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile sent message    {"request": {"setCommit":{"branch":{"name":"master","repo":{"name":"images","project":{"name":"default"},"type":"user"}},"repo":{"name":"images","project":{"name":"default"},"type":"user"}}}}
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile sent message    {"request": {"addFile":{"path":"/test.txt","raw":"aGVsbG8K"}}}
All messages sent
2023-08-31T14:01:51.724-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile send side of stream closed
2023-08-31T14:01:51.765-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile server headers  {"headers": {"content-type": "application/grpc", "date": "Thu, 31 Aug 2023 18:01:51 GMT", "server": "envoy", "x-envoy-upstream-service-time": "40"}}
2023-08-31T14:01:51.765-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile received message        {"reply": {}}
2023-08-31T14:01:51.766-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile pfs_v2.API/ModifyFile: span finished ok {"spanDuration": "41.782179ms", "trailer": {}}
{}
2023-08-31T14:01:51.766-0400    DEBUG   grpcClient.stream.pfs_v2.API/ModifyFile pfs_v2.API/ModifyFile: span finished ok {"spanDuration": "41.802679ms", "trailer": {}}
...
```

### CODE REVIEW

File: src/internal/log/field_test.go [hunk 1]
```diff
@@ -54,4 +57,17 @@ func TestProto(t *testing.T) {
 
 	Info(ctx, "time", Proto("time", timestamppb.Now()))
 	Info(ctx, "nil time", Proto("time", (*timestamppb.Timestamp)(nil)))
+
+	v := &versionpb.Version{}
+	dynamic := dynamicpb.NewMessage(v.ProtoReflect().Descriptor())
+	dynamic.Set(v.ProtoReflect().Descriptor().Fields().ByName("major"), protoreflect.ValueOf(uint32(3)))
```
Comment (brendoncarroll) [inline src/internal/log/field_test.go:63]: Where does this 3 come from?

Comment (jrockway) [reply src/internal/log/field_test.go:63]: Oh, just some random number created for testing.  `2` is too predictable for that field.

File: src/internal/log/field_test.go [hunk 2]
```diff
@@ -54,4 +57,17 @@ func TestProto(t *testing.T) {
...
+	raw, err := protoToJSONMap(dynamic)
+	if err != nil {
+		t.Errorf("protoToJSONMap: %v", err)
+	}
+	if diff := cmp.Diff(map[string]interface{}{"major": float64(3)}, raw); diff != "" {
```
Comment (brendoncarroll) [inline src/internal/log/field_test.go:68]: nit: `interface{}` -> `any`, since you used it in the other file.

File: src/server/misc/cmds/grpc.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		license.File_license_license_proto,
+		pfs.File_pfs_pfs_proto,
+		pps.File_pps_pps_proto,
+		proxy.File_proxy_proxy_proto,
+		transaction.File_transaction_transaction_proto,
+		versionpb.File_version_versionpb_version_proto,
+		worker.File_worker_worker_proto,
+	)
+
+	// If no args, print all available RPCs.  The names can be difficult to
```
Comment (brendoncarroll) [inline src/server/misc/cmds/grpc.go:59]: :+1:

**Review by brendoncarroll — APPROVED** [review]

File: src/internal/log/field_test.go (hunk 2 above)
Comment (jrockway) [reply src/internal/log/field_test.go]: Done.

**Pull request merged**